packages/client/e2e/test-f.spec.ts

```diff
@@ -0,0 +1,532 @@
+    await expect(miniPlayerWithRoomName).toBeVisible();
+
+    await miniPlayerWithRoomName.click();
+}
+
+async function waitForTrackCardToHaveScore({
+    page,
+    score,
+    trackID,
+}: {
+    page: Page;
+    score: string;
+    trackID: string;
+}) {
+    const suggestedTrackScore = page.locator(
+        `css=:text("${score}"):right-of([data-testid="${trackID}-track-card"])`,
+    );
+    await expect(suggestedTrackScore).toBeVisible({
+        timeout: 10_000,
+    });
+}
+
+test.afterEach(async ({ browser }) => {
+    await closeAllContexts(browser);
+});
+
+test('Test F', async ({ browser }) => {
+    const [
+        { page: userADevice1Page },
+        { context: userBContext, page: userBDevice1Page },
+    ] = await Promise.all([
+        setupAndGetUserPage({
+            browser,
+            userIndex: 0,
+            knownSearches,
+            town: 'Manosque, France',
+        }),
+        setupAndGetUserPage({
+            browser,
+            userIndex: 1,
+            knownSearches,
+            town: 'Paris, France',
+        }),
+    ]);
+
+    await userBDevice1Page.waitForTimeout(1_000);
+    const { page: userBDevice2Page } = await createNewTabFromExistingContext(
+        userBContext,
+    );
+
+    const roomName = 'MusicRoom is the best';
+
+    const { initialTrack, initialTrackID } =
+        await createPublicRoomWithTimeAndPhysicalConstraints({
+            page: userADevice1Page,
+            roomName,
+            address: 'Manosque, France',
+            // startsAt: addMinutes(new Date(), 1),
+            startsAt: new Date(),
+            endsAt: addHours(new Date(), 6),
+        });
+
+    await joinRoom({ page: userBDevice1Page, roomName });
+
+    await openFullScreenPlayer({
+        page: userBDevice2Page,
+        roomName,
+    });
+
+    // User B votes for the initial track.
+    // It should have no impact.
+    // If it has one, the rest of the test will break.
+    await voteForTrackInMusicPlayerFullScreen({
+        page: userBDevice1Page,
+        trackToVoteFor: initialTrack,
+    });
+
+    // We wait for the room to allow votes according to time constraints.
+    // await userBDevice1Page.waitForTimeout(60_000);
+
+    // User B votes again for the initial track.
+    // It should fail as the user does not yet match
+    // physical constraints.
+    await voteForTrackInMusicPlayerFullScreen({
+        page: userBDevice1Page,
+        trackToVoteFor: initialTrack,
+    });
+
+    // User B enters in room area.
+    await userBContext.setGeolocation(
+        GEOLOCATION_POSITIONS['Manosque, France'],
+    );
+    await userBDevice1Page.reload();
+    await initPage(userBDevice1Page);
+    await openFullScreenPlayer({
+        page: userBDevice1Page,
+        roomName,
+    });
+
+    await voteForTrackInMusicPlayerFullScreen({
+        page: userBDevice1Page,
+        trackToVoteFor: initialTrack,
+    });
+
+    await Promise.all([
```

Comment:
Could this promise.all and the following one be one ?

Reply:
Yep!
It's been implemented and tested in the last commit I just pushed.

Reply:
Great !